### ISSUE TITLE
Services setup guide?

### ISSUE THREAD

**Issue by razordead** [fullphat/snarl#29]
Would it be possible to get setup guides for Services?  I'm primarily interested in Instapush.  I can't seem to connect Snarl R5 beta 2 to my Instapush account.

**Comment by fullphat** [comment]
Just had a quick look and it seems I've managed to gnaws up the UI.  I'll get you some workaround instructions asap and upload a fixed version.

**Comment by fullphat** [comment]
Update: tested and working! :)

1. Log in to instapush (via browser, not app)
2. Click on "info" and copy your user token
3. Go to Snarl management console
4. Services > Instapush
5. Click "Settings"
6. Click "Add new Scheme"
7. Give it a name 
8. Paste your user token in the "Access Token" field
9. Save settings
10. Create a new rule with Instapush:{your scheme} as the destination
11. Click "Test" on the rule page

Does this work?

**Comment by razordead** [comment]
That didn't work for Instapush, but it worked for Pushbullet.
[http://instapush.im/developer/rest#auth](http://instapush.im/developer/rest#auth) seems to indicate that the user token is only used for listing & adding apps.  The application-id & application-secret seem to be needed for sending notifications.

**Comment by fullphat** [comment]
Just had chance to refresh my memory.  If you enter a valid user token, Snarl _should_ create an application within Instapush which it will then use for notifications.  Can you check your Instapush dashboard to see if it's managed to do that?  I will rework the service to require `app-id` and `app-secret` anyway I think, as that's much easier - and more flexible for the user...

**Comment by fullphat** [comment]
I've uploaded a special Beta 3 release here: https://sourceforge.net/projects/snarlwin/files/Snarl/R5/?upload_just_completed=true 

Can you test it out?  I've changed the way the Instapush redirect works so now you need to create the app and events you want within Instapush itself.

To configure Snarl, create a new Instapush scheme (this beta will ignore any existing schemes you have created, sorry), enter the `app-id` and `app-secret` for your Instapush app and the `event` to use.

Snarl currently exports the following trackers:

```
[title]
[text]
[callback]
[machine]
[source]
[app]

```
All should be pretty self-explanatory.  If any are null, the redirect will send a single space to prevent Instapush from complaining about a missing tracker.

Let me know if you'd like other trackers exporting and I'll see what I can do.

Hope this all makes sense! 😀

**Comment by razordead** [comment]
Yeah, when using my user token, Snarl never managed to create an app inside Instapush.  Is there something specific I need to do on the Snarl side to get it to register, or just send a test notification?
Using the special Beta 3, manually creating an app in Instapush & entering it's app-id & app-secret in Snarl, Instapush still doesn't receive test notifications from Snarl.

**Comment by fullphat** [comment]
How are you testing notifications using the special Beta 3?
